src/foam/nanos/export/GoogleSheetsExportDriver.js

```diff
@@ -57,18 +79,30 @@ foam.CLASS({
 
       var columnConfig = X.columnConfigToPropertyConverter;
 
-      var propNames = X.filteredTableColumns ? X.filteredTableColumns : this.outputter.getAllPropertyNames(dao.of);
+      var propNames;
+      if ( ! this.template )
+        propNames =  X.filteredTableColumns ? X.filteredTableColumns : this.outputter.getAllPropertyNames(dao.of);
+      else {
+        var expr1 = foam.mlang.Expressions.create();
+        var template = await X.reportTemplateDAO.find(expr1.EQ(foam.nanos.export.report.Template.ID, this.template));
+        propNames = template && template.columnNames && template.columnNames.length > 0 ? template.columnNames : X.filteredTableColumns ? X.filteredTableColumns : this.outputter.getAllPropertyNames(dao.of);
+      }
+       
       propNames = columnConfig.filterExportedProps(dao.of, propNames);
 
       var metadata = await self.outputter.getColumnMethadata(X, dao.of, propNames);
 
+      // to backend
```

Comment:
I think that it would be better to do projection on backend so we don't send it back and forth as we need this data only on backend, but we do formatting on front-end so I will add java version of unitPropValueToString.

Reply:
or should I just override toString for those classes?